src/xcode/ENA/ENA/Source/Scenes/ENSetting/ENStateHandler.swift

```diff
@@ -35,14 +35,22 @@ class ENStateHandler {
 			stateDidChange()
 		}
 	}
-
+	
+	private let reachabilityService: ReachabilityService
 	private weak var delegate: StateHandlerObserverDelegate?
 	private var internetOff = false
 
-	init(_ initialState: ExposureManagerState, delegate: StateHandlerObserverDelegate) {
+	init(
+		_ initialState: ExposureManagerState,
+		reachabilityService: ReachabilityService,
+		delegate: StateHandlerObserverDelegate
+	) {
+		self.reachabilityService = reachabilityService
 		self.delegate = delegate
 		currentState = determineCurrentState(from: initialState)
-		try? addReachabilityObserver()
+		self.reachabilityService.observe(on: self) { [weak self] reachabilityState in
+			self?.internet(reachabilityState == .connected)
```

Comment:
This line seems not to be called when changing the disconnecting internet. Can you check again?

Reply:
@steins-code sorry my fault 😄  I've just added a missing invocation for the `setup` function https://github.com/corona-warn-app/cwa-app-ios/pull/53/commits/1ff739bf7b67039690225677f8b5f85446e43992 please try again.

Reply:
Thanks for your efforts :)